src/main/java/net/dries007/tfc/ForgeEventHandler.java

```diff
@@ -637,6 +642,33 @@ public static void onPlayerTick(TickEvent.PlayerTickEvent event)
         {
             foodData.addThirst(TFCConfig.SERVER.thirstGainedFromDrinkingInTheRain.get().floatValue());
         }
+
+        // todo: PotionExpiryEvent is not called on the client due to bad design. So we have to fake it here. Needs a forge PR
```

Comment:
Can we not send a packet from S -> C that just triggers the effect client side? Same as we do i.e. to detect when a player drinks (clientside event which sends C -> S to do logic)